### ISSUE TITLE
Create ExtendedField rows when creating FacilityMatch rows

### ISSUE THREAD

**Issue by jwalgran** [open-apparel-registry/open-apparel-registry#1493]
## Overview

Any time we parse a `FacilityListItem` (via list or via API), we should create `ExtendedField`s. We should create `ExtendedField`s when submitting facility claim details. 

Depends on #1492

### Describe the solution you'd like

- Update the `ExtendedField` model to allow null for `facility`

- When parsing a FacilityListItem, look for well known extended field keys in the submitted list item and create `ExtendedField` rows. The `facility` field will be null. Update the `facility` when the `FacilityListItem`s facility is updated.
- When a moderation activity changes the facility on a facility list item, also update `ExtendedFields`s (when promoting a facility list item)

- When `post`ing to `get_claimed_details`, look for well known extended field keys and create_or_update  `ExtendedField`s.  
     - Loop through known claim fields that correlate to known ExtendedFields. If a value is present and not blank/empty, create_or_update an ExtendedField. If it is blank or missing, look for existing ExtendedFields for the given claim for the field and delete them. 

- Create a follow-up card to add `ExtendedField` deletion when deleting a facility.
